File: scheduler/src/main/java/ch/fhnw/edu/eaf/scheduler/domain/Event.java
package ch.fhnw.edu.eaf.scheduler.domain;

import java.util.Collection;
import java.util.Date;

/**
 * An event-object
 */
public class Event {

    /**
     * The id of an event.
     */
    public Long id;

    /**
     * Date when the event was created.
     */
    public Date created;

    /**
     * Date when the event was last updated.
     */
    public Date updated;

    /**
     * The name/title of the event
     */
    public String name;

    /**
     * The description of the event.
     */
    public String description;

    /**
     * The location of the event.
     */
    public String location;

    /**
     * The time when the event starts.
     */
    public Date startTime;

    /**
     * The time when the event is "closed", that is the registration-deadline for the event.
     */
    public Date closingTime;

    /**
     * The time when the event finishes.
     */
    public Date endTime;

    /**
     * Indicates whether the "deadline"-Mails to the sv-group, referenten and raumkoordination were sent.
     */
    public boolean closingMailSend;

    /**
     * Files accompanying this event.
     */
    public File image;

    public Collection<File> files;
    public Collection<User> speakers;
    public Collection<EventAttendee> attendees;

}

File: ionic/src/models/Event.ts
export class Event {
  /**
   * The datetime when an event-deadline closes.
   */
  closingTime: string;

  /**
   * The date when an event was created.
   */
  created: string;

  /**
   * The descripition of an event.
   */
  description: string;

  /**
   * The location where the event will take place.
   */
  location: string;

  /**
   * Whether or not the automatically-to-send-mails were already sent.
   */
  closingMailSend: boolean;

  /**
   * The time when an event finishes.
   */
  endTime: string;

  /**
   * The id of an event.
   */
  id: number;

  /**
   * The title of an event.
   */
  name: string;

  /**
   * The image accompanying the event.
   */
  image: string;

  /**
   * The url to the image of the event.
   */
  imageUri: string;

  /**
   * The datetime when an event starts.
   */
  startTime: string;

  /**
   * The date when an event was last updated.
   */
  updated: string;

  /**
   * A list of all speakers that will speak at the event.
   */
  speakers: string[];
}
